### PR TITLE
swift 5

### PR DESCRIPTION
**Pull request by pranjalsatija**

### CODE REVIEW

**Comment by pranjalsatija** [comment]
can confirm that on my machine:
* the project builds
* the example runs
* the tests pass

File: Tests/DirectoryTests.swift [hunk 1]
```diff
@@ -190,11 +189,11 @@ class DirectoryTests: XCTestCase {
         let textExpectation = expectation(description: "testRead")
         
         let textA = "This is a test read A"
-        try? textA.data(using: .utf8)?.write(to: cacheDirectory.appendingPathComponent("testReadA", isDirectory: false))
+        ((try? textA.data(using: .utf8)?.write(to: cacheDirectory.appendingPathComponent("testReadA", isDirectory: false))) as ()??)
```
Comment (tylermilner) [inline Tests/DirectoryTests.swift]: Let's see if we can fix these double optionals (I assume the migrator did this).

Comment (earlgaspard) [reply Tests/DirectoryTests.swift]: It was the migrator.

**Comment by earlgaspard** [comment]
the .travis.yml needs to be updated to 10.2
`osx_image: xcode10.2`

File: Tests/DirectoryTests.swift (hunk 1 above)
Comment (pranjalsatija) [reply Tests/DirectoryTests.swift]: fixed

**Comment by wmcginty** [comment]
Just needs a Changelog entry @pranjalsatija and it looks good.

**Comment by tylermilner** [comment]
@wmcginty are we planning to release this in its current state as a `1.0` release? I know there was a rewrite happening at some point - do we still plan on doing that? If so, I'd advocate for not doing a `1.0` release until the rewrite is complete. Less migration docs to write! 😅

**Comment by wmcginty** [comment]
@tylermilner I don't have time to at the moment, but I know @pranjalsatija was interested in taking that work on. If that's the case we can just use `Master` to start recording changes leading up to 1.0?

**Comment by pranjalsatija** [comment]
@wmcginty changelog has been added

@tylermilner I added it anyway, but I'm starting the rewrite as well.

**Pull request merged**